src/cockroach_db/mod.rs

```diff
@@ -0,0 +1,121 @@
+use std::collections::BTreeMap;
+
+use testcontainers::{core::WaitFor, Image, ImageArgs};
+
+const DEFAULT_IMAGE_NAME: &str = "cockroachdb/cockroach";
+const DEFAULT_IMAGE_TAG: &str = "latest-v23.2";
+
+/// Module to work with [`Cockroach DB`] inside of tests.
+///
+/// This module is based on the official [`Cockroach docker image`].
+///
+/// # Example
+/// ```
+/// use testcontainers::clients;
+/// use testcontainers_modules::cockroach_db;
+///
+/// let docker = clients::Cli::default();
+/// let cockroach = docker.run(cockroach_db::CockroachDb::default());
+///
+/// let http_port = cockroach.get_host_port_ipv4(26257);
+///
+/// // do something with the started cockroach instance..
+/// ```
+///
+/// [`Cockroach`]: https://www.cockroachlabs.com/
+/// [`Cockroach docker image`]: https://hub.docker.com/r/cockroachdb/cockroach
+/// [`Cockroach commands`]: https://www.cockroachlabs.com/docs/stable/cockroach-commands
+#[derive(Debug)]
+pub struct CockroachDb {
+    name: String,
+    tag: String,
+    env_vars: BTreeMap<String, String>,
+}
+
+impl Default for CockroachDb {
+    fn default() -> Self {
+        CockroachDb::new(
+            DEFAULT_IMAGE_NAME.to_string(),
+            DEFAULT_IMAGE_TAG.to_string(),
+        )
+    }
+}
+
+impl CockroachDb {
+    fn new(name: String, tag: String) -> Self {
+        CockroachDb {
+            name,
+            tag,
+            env_vars: Default::default(),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct CockroachDbArgs {
+    pub command: String,
+    pub args: Vec<String>,
```

Comment:
I'd better encapsulate them. It will be easier to change without breaking changes.

For example, add `new` which accepts command with arguments and returns the instance. 

And inside we can even store only 1 `Vec`

Reply:
I think encapsulating them is a good idea, but I prefer to keep the strings separate

Reply:
Done